Battery monitoring fixes and improvements

BrainFPV opened the pull request:
Improvements to the battery monitoring module:

- Unchecking the voltage/current monitoring checkbox now actually disables voltage/current measurements. Before, it was necessary to set the ADC pin to `NONE` for this. Which is unintuitive and can lead to the problem that current monitoring remains enabled even when there is no sensor attached.

- Makes the flight time thresholds configurable

-  Shows consumed capacity and remaining flight time in battery monitoring config

- Moves ADC pin selection to the voltage / current config groups in the GCS

BrainFPV commented:
Please review, should be ready.

peabody124 commented:
I'm wondering if the "SensorType" field in ```FlightBatterySettings``` provides any use beyond setting the Current/Voltage pins to none or not. Why have this redundancy?

BrainFPV commented:
Yes, it's completely redundant and could be removed. At first I wasn't even avware that it is there and changed the GCS code to set the pin to NONE when the checkbox was unchecked. Maybe this would be the best solution, i.e., still have the checkboxes but unchecking them sets the pin to NONE, checking them sets the pin to the default value. Would definitely be cleaner. Thoughts?

BrainFPV commented:
Also, the `Type` and `NbCells` fields are unused and could be removed. I assume the idea was to use them to determine the voltage thresholds but it was never implemented. It would make more sense to use this information in the GCS and set the thresholds accordingly (e.g. using a wizard). There is no need for the FC to know this information.

peabody124 commented:
Yeah I think dropping all that redundancy would be great. If you want to do it for this PR that would be preferable but if not this improvement can also stand alone.

BrainFPV commented:
I will do it, pruning things now makes sense. Stay tuned.

fujin commented:
Looking forward to this!

lolorc commented:
Hi there,

I have a related use case, I'm using a Frsky FLVSS voltage monitor. It's connected to the S.PORT bus with the FC and the X4R. In my case the battery module is disabled so the s.port module doesn't send the cells/voltage s.port fields.

It works as expected, I get both voltage and TL telem on my TX.

I was thinking of implementing the reading of the s.port bus from the s.port module so TL can make use of the sensors on the s.port bus.

would that be ok to have the battery monitoring module to use the voltage from s.port instead of an ADC ?
would that be ok to have the voltage/current sources individually selectable in the gui ? (ADC/SPORT) ?

It would need another PR, I'm just wondering if that's a valid use case.

peabody124 commented:
Yeah you could add a new source instead of ADC# to get the battery / voltage signal from. Then you'd need to add support in the SPort code. The last thing would be how do you transport the data. I'd say the SPORT driver should expose a PIOS_SPORT_GetBattery() and PIOS_SPORT_SetBattery() methods.

BrainFPV commented:
@peabody124 I removed the unused fields from settings (`SensorType` and `Type`). `NbCells` is used by the `UAVOFrSKYSensorHubBridge` so I left it there. I also added simplified the battery monitoring code and added some logic to the GCS so that unchecking the voltage/current checkbox sets the voltage/current pins to `NONE` (and the other way around).

BrainFPV commented:
Please review

peabody124 commented:
Tested and works well. Good stuff!

BrainFPV commented:
Thanks :)
